### PR TITLE
changed add_javescript to add_js_file

### PR DESCRIPTION
**Pull request by shimwell**
As reported in 
#20 

I noticed the error report in a recent build of using this package had the error 
```
AttributeError: 'Sphinx' object has no attribute 'add_javascript'
```
Screenshot from 2021-05-09 23-30-03

Then I found the deprecated API infor in Sphinx than mentioned ```add_javascript()``` has been replaced by ```add_js_file()``` and will be removed in v4.0. 

https://www.sphinx-doc.org/en/master/extdev/deprecated.html?highlight=add_javascript#deprecated-apis

Screenshot from 2021-05-09 23-32-33

So I've replaced occurrences of add_javascript with add_js_file in the sphinxcadquery/sphinxcadquery.py file

I think this will continue to work with old version and new versions of Sphinx. But I've not found a nice way to test it.

### CODE REVIEW

**Comment by Peque** [comment]
@Shimwell Thanks for the pull request and for making some investigations on the issue! :blush: 

As you may have already noticed, there is no CI setup for this project (no tests), so could you please test the changes locally to verify everything works as expected? If you test it with Sphinx 4 only, that would be fine. If you test it with Sphinx 3 too, that would be perfect. :stuck_out_tongue_winking_eye: 

I suspect more changes may be needed for Sphinx 4 compatibility. In example, `add_stylesheet` seems that it was also deprecated in 1.8 and removed in 4.0.

File: sphinxcadquery/sphinxcadquery.py [hunk 1]
```diff
@@ -159,14 +159,14 @@ def copy_asset_files(app, exc):
...
-    app.add_javascript('sphinxcadquerystatic/main.js')
+    app.add_js_file('sphinxcadquerystatic/three.js')
+    app.add_js_file('sphinxcadquerystatic/AMFLoader.js')
+    app.add_js_file('sphinxcadquerystatic/STLLoader.js')
+    app.add_js_file('sphinxcadquerystatic/LegacyJSONLoader.js')
+    app.add_js_file('sphinxcadquerystatic/jszip.min.js')
+    app.add_js_file('sphinxcadquerystatic/OrbitControls.js')
+    app.add_js_file('sphinxcadquerystatic/WebGL.js')
+    app.add_js_file('sphinxcadquerystatic/main.js')
     app.add_stylesheet('sphinxcadquerystatic/main.css')
```
Comment (Peque) [inline sphinxcadquery/sphinxcadquery.py]: I suspect `add_stylesheet` will also throw an error. Can you test everything is working as expected locally before we merge this? :blush:

**Review by Peque — CHANGES_REQUESTED** [review]

File: sphinxcadquery/sphinxcadquery.py (hunk 1 above)
Comment (shimwell) [reply sphinxcadquery/sphinxcadquery.py]: Thanks for the hints on the issue, this builds docs locally now. You are right it also required add_stylesheet -> add_css_file .

**Comment by Peque** [comment]
@Shimwell Looking good! :blush: 

Would you mind squashing the changes into a single commit with title "Fix Sphinx 4 compatibility issues"?

**Review by Peque — APPROVED** [review]

**Comment by shimwell** [comment]
> @Shimwell Looking good! blush
> 
> Would you mind squashing the changes into a single commit with title "Fix Sphinx 4 compatibility issues"?

Not sure if I got that correct (first time I've tried a git squash) but I shall make a cleaner PR and then you can take your pick :-)

**Comment by Peque** [comment]
@Shimwell Don't worry, I can do that for you if you want. If you'd rather try again yourself, I'll wait! :blush: 

By the way, I meant "Fix Sphinx 4 compatibility issues", but without the quotes around. :stuck_out_tongue_winking_eye:

**Comment by shimwell** [comment]
> @Shimwell Don't worry, I can do that for you if you want. If you'd rather try again yourself, I'll wait! blush
> 
> By the way, I meant "Fix Sphinx 4 compatibility issues", but without the quotes around. stuck_out_tongue_winking_eye

Yep I wasn't sure about those quote marks were needed in the Nano commit message. Thanks for training me up here.

**Comment by Peque** [comment]
Closing after https://github.com/Peque/sphinxcadquery/pull/25.

**Pull request closed without merging**